processing/src/main/java/io/druid/query/timeboundary/TimeBoundaryQueryQueryToolChest.java

```diff
@@ -152,8 +137,10 @@ public boolean apply(T input)
       public byte[] computeCacheKey(TimeBoundaryQuery query)
       {
         final byte[] cacheKey = query.getCacheKey();
-        return ByteBuffer.allocate(1 + cacheKey.length)
+        final byte descending = query.isDescending() ? (byte)1 : 0;
+        return ByteBuffer.allocate(2 + cacheKey.length)
```

Comment:
does it affect the result though? if descending doesn't affect timeboundary results, we probably don't need to have a flag in the cache for it.